app/run.py

```diff
@@ -153,9 +153,9 @@ def callback(pubsub_message):
             resource_creds = cb.get_credentials(
                 **resource.to_dict(),
```

Comment:
The call to `to_dict()` here results in the resource creating its client. The setter for the `client_kwargs` property clears the client so the next time its instantiated it has the updated kwargs. Since the previous implementation updated the dictionary directly, the setter was never called, and the credentials were never updated.

So now we trigger the setter below to resolve the issue